test-skripsi/test-prod.tsx

```diff
@@ -10,27 +17,33 @@ class InventoryManager {
 
 	constructor() {
 		this.inventory = [];
+		this.loadInventory();
+	}
+
+	private findProductById(productId: number): Product | undefined {
+		return this.inventory.find((p) => p.id === productId);
 	}
 
-	addProduct(product: Product) {
-		const index = this.inventory.findIndex((p) => p.id === product.id);
-		if (index !== -1) {
-			console.log("Product already exists.");
-			return;
+	addProduct(product: Product): boolean {
+		if (this.findProductById(product.id)) {
+			throw new InventoryError("Product already exists.");
```

Comment:
Throwing an error for an existing product might be too harsh for some use cases. Consider if a softer handling or a boolean return value might be more appropriate in some contexts.